serqlane/astcompiler.py

```diff
@@ -1010,9 +1043,9 @@ def handle_literal(self, tree: Tree, expected_type: Type, lookup_name: str, lite
         if literal_kind is TypeKind.literal_string:
             value = self.resolve_escape_sequence(value)
 
-        if expected_type is not None:
+        if expected_type != None:
```

Comment:
this is not how none should be checked for

Reply:
this is not resolved; please see <https://peps.python.org/pep-0008/#programming-recommendations>
```
Comparisons to singletons like None should always be done with is or is not, never the equality operators.
```